SerialPrograms/Source/PokemonSV/Programs/FastCodeEntry/PokemonSV_FastCodeEntry.cpp

```diff
@@ -125,6 +126,14 @@ void FastCodeEntry::program(MultiSwitchProgramEnvironment& env, CancellableScope
         return;
     }
 
+    if (MODE == Mode::MYSTERY_GIFT){
+        const char* error = enter_code(env, scope, settings, CODE, true, true);
+        if (error){

```

Comment:
I don't think this if can ever happen as you always return an empty string

Reply:
merged the logic with the normal mode check and have error report only if on normal mode